calliope/core/model.py

```diff
@@ -301,6 +301,23 @@ def _generate_default_dict(self) -> AttrDict:
             }
         )
 
+    def _add_run_mode_custom_math(self) -> None:
+        """If not given in the custom_math list, override model math with run mode math"""
+        run_mode = self.run_config["mode"]
+        # FIXME: available modes should not be hardcoded here.
+        # They should come from a YAML schema.
+        not_run_mode = {"plan", "operate", "spores"}.difference([run_mode])
+        run_mode_mismatch = not_run_mode.intersection(self.model_config["custom_math"])
+        if run_mode_mismatch:
```

Comment:
This extra check suggests to me that there should be two separate directories for math: one with a file for each run mode and one with custom math. Then you don't need to do this extra checking. Or you could check if a run mode was defined that doesn't have a corresponding file in the run mode math directory.

Reply:
The check here is actually for whether some run mode custom math has been applied by the user explicitly, which I think is a valid use case if you want to control in what order that custom math appears (otherwise it comes at the end and cannot be overridden). You may also want to apply the custom math of an in-built run mode so you can have it available for your custom run mode. 

I'm also not keen on creating more sub-directories if it can be helped.

Agreed though that this would be better if not hardcoded here. It would also be great if the run mode custom math wasn't linked to its custom math by filename alone. Ideally, this link would be in some config file instead and would allow a run mode in future to e.g., depend on multiple custom math files.

Reply:
Ok, let's park it for now, and keep it hardcoded as-is with the FIXME marker.